Svc/ComQueue/ComQueue.cpp

```diff
@@ -154,12 +159,21 @@ void ComQueue::comStatusIn_handler(const FwIndexType portNum, Fw::Success& condi
         // On success, the queue should be processed. On failure, the component should still wait.
         case WAITING:
             if (condition.e == Fw::Success::SUCCESS) {
+#if FW_COM_BUFFER_RETRY_ON_FAILURE
+                this->m_hasRetried = false;
+#endif
                 this->m_state = READY;
                 this->processQueue();
                 // A message may or may not be sent. Thus, READY or WAITING are acceptable final states.
                 FW_ASSERT((this->m_state == WAITING || this->m_state == READY),
                           static_cast<FwAssertArgType>(this->m_state));
             } else {
+#if FW_COM_BUFFER_RETRY_ON_FAILURE
+                if (!this->m_hasRetried) {
```

Comment:
We technically should only send a retry when "SUCCESS" is received.  When a failed message comes in, we should set a `m_need_retry` boolean here. 

If `m_need_retry` is set we should read your member variable and send there.  I'll leave that note there too.